src/js/user-profile/reducers/profile.js

```diff
@@ -35,8 +35,8 @@ const initialState = {
 
 function profileInformation(state = initialState, action) {
   switch (action.type) {
-    case UPDATE_PROFILE_FIELD: {
-      return _.set(action.propertyPath, action.value, state);
+    case UPDATE_PROFILE_FIELDS: {
+      return _.merge(state, action.newState);
```

Comment:
I'm not sure merge is correct, I think you want to do `assign`. You could get old and new array items mixed together with merge, I think.